beacon/goclient/goclient_test.go

```diff
@@ -17,54 +22,352 @@ import (
+				},
+				delay: 2 * time.Second,
+			},
+		}
+
+		err := runHealthyTest(t, syncResponseList, 2, false)
+		require.ErrorIs(t, err, context.DeadlineExceeded)
+	})
+
+	t.Run("multi client: first times out", func(t *testing.T) {
+		t.Parallel()
+
+		syncResponseList := []syncResponse{
+			{
+				state: &v1.SyncState{
+					SyncDistance: phase0.Slot(0),
+					IsSyncing:    false,
+				},
+				delay: 2 * time.Second,
+			},
+			{
+				state: &v1.SyncState{
+					SyncDistance: phase0.Slot(0),
+					IsSyncing:    false,
+				},
+			},
+		}
+
+		err := runHealthyTest(t, syncResponseList, 2, false)
+		require.NoError(t, err)
+	})
+
+	t.Run("multi client: second times out", func(t *testing.T) {
+		t.Parallel()
+
+		syncResponseList := []syncResponse{
+			{
+				state: &v1.SyncState{
+					SyncDistance: phase0.Slot(0),
+					IsSyncing:    false,
+				},
+			},
+			{
+				state: &v1.SyncState{
+					SyncDistance: phase0.Slot(0),
+					IsSyncing:    false,
+				},
+				delay: 2 * time.Second,
+			},
+		}
+
+		err := runHealthyTest(t, syncResponseList, 2, false)
+		require.NoError(t, err)
+	})
+
+	t.Run("multi client: concurrent check, both synced", func(t *testing.T) {
+		t.Parallel()
+
+		syncResponseList := []syncResponse{
+			{
+				state: &v1.SyncState{
+					SyncDistance: phase0.Slot(0),
+					IsSyncing:    false,
+				},
+			},
+			{
+				state: &v1.SyncState{
+					SyncDistance: phase0.Slot(0),
+					IsSyncing:    false,
+				},
+				delay: 2 * time.Second,
+			},
 		}
 
-		err := runHealthyTest(t, syncData, 3)
+		err := runHealthyTest(t, syncResponseList, 2, true)
 		require.NoError(t, err)
 	})
+
+	t.Run("multi client: concurrent check, only first synced", func(t *testing.T) {
+		t.Parallel()
+
+		syncResponseList := []syncResponse{
+			{
+				state: &v1.SyncState{
+					SyncDistance: phase0.Slot(0),
+					IsSyncing:    false,
+				},
+			},
+			{
+				state: &v1.SyncState{
+					SyncDistance: phase0.Slot(3),
+					IsSyncing:    true,
+				},
+			},
+		}
+
+		err := runHealthyTest(t, syncResponseList, 2, true)
+		require.NoError(t, err)
+	})
+
+	t.Run("multi client: concurrent check, only second synced", func(t *testing.T) {
+		t.Parallel()
+
+		syncResponseList := []syncResponse{
+			{
+				state: &v1.SyncState{
+					SyncDistance: phase0.Slot(3),
+					IsSyncing:    true,
+				},
+			},
+			{
+				state: &v1.SyncState{
+					SyncDistance: phase0.Slot(0),
+					IsSyncing:    false,
+				},
+			},
+		}
+
+		err := runHealthyTest(t, syncResponseList, 2, true)
+		require.NoError(t, err)
+	})
+
+	t.Run("multi client: concurrent check, no synced", func(t *testing.T) {
+		t.Parallel()
+
+		syncResponseList := []syncResponse{
+			{
+				state: &v1.SyncState{
+					SyncDistance: phase0.Slot(3),
+					IsSyncing:    true,
+				},
+			},
+			{
+				state: &v1.SyncState{
+					SyncDistance: phase0.Slot(4),
+					IsSyncing:    true,
+				},
+			},
+		}
+
+		err := runHealthyTest(t, syncResponseList, 2, true)
+		require.ErrorIs(t, err, errSyncing)
+	})
+}
```

Comment:
i think it would be better to refactor this to testCases, something like this:
```go
testCases := []struct {
		name                  string
		syncResponseList      []syncResponse
		concurrentHealthCheck bool
		wantErr               error
	}{
```
less code duplication + more readable structure